### DISCUSSION TITLE
Is EXPORT DATABASE transactional?

### DISCUSSION THREAD

**Discussion by chairmank** [duckdb/duckdb#6865] [Q&A] [answered]
The [current documentation](https://duckdb.org/docs/sql/statements/export.html) for `EXPORT DATABASE` does not describe how this operation interacts with concurrent writes to the database.

* Does `EXPORT DATABASE` block concurrent writes to the database while the operation is in progress?
* If concurrent writes are allowed, does `EXPORT DATABASE` guarantee a consistent snapshot of the database?

**Comment by Mytherin** [comment, marked as answer]
`EXPORT DATABASE` operates inside the transaction of the client that sends the query (like all queries in DuckDB). Clients are guaranteed to always see a consistent snapshot of the database due to the MVCC snapshot isolation that DuckDB provides. Concurrent writes are allowed while `EXPORT DATABASE` is running - these will then not be part of the snapshot the current client is viewing and hence will be ignored/not exported.
